src/main/java/org/fenixedu/academic/domain/Degree.java

```diff
@@ -316,6 +316,29 @@ public DegreeCurricularPlan createPreBolonhaDegreeCurricularPlan(String name, De
         }
     }
 
+    public DegreeCurricularPlan createDegreeCurricularPlan(String name, GradeScale gradeScale, Person creator) {
```

Comment:
With this change, won't the code in `createBolonhaDegreeCurricularPlan` be duplicated here? If so, in `createBolonhaDegreeCurricularPlan`, in the `isBolonhaDegree` case, just invoke this method.